bugcrowd/vulnerability-rating-taxonomy
# [BC-17151] As a template contributor, when I add a new VRT template, …

### User story
As a template contributor
When I add a new VRT template
I should have tests to verify that my templates map to an existing VRT item
So that i can map newly added templates

Jira ticket [BC-17151](https://bugcrowd.atlassian.net/browse/BC-17151)

### Technical explanation

Added VRT mapping to the templates
Added `template.json` inside the `mappings` folder
Added test case for validating mapping url is exist or not


### To test
1. Run the below commands in your terminal
```
pip3 install -r lib/requirements.txt
python3 -B lib/validate_vrt.py
```

2. You can see the valid results
<img width="1141" alt="Screenshot 2021-09-15 at 7 03 43 PM" src="https://user-images.githubusercontent.com/57894121/133443045-2adda8ec-3dd9-47b2-8406-f52e942daebd.png">

3. To test the negative scenarios
In `template.json` file just modify the template value as invalid

```
{
  "metadata": {
    "default": null
  },
  "content": [
    {
      "id": "server_security_misconfiguration",
      "children": [
        {
          "id": "unsafe_cross_origin_resource_sharing",
          "template": "template.testing"
        }
     ]
   }]
}
```
4. Again run the `python3 -B lib/validate_vrt.py`
5.  You can see the error message for template missing
<img width="1214" alt="Screenshot 2021-09-15 at 7 08 49 PM" src="https://user-images.githubusercontent.com/57894121/133443999-f8c653b8-9606-463e-b18c-26b715adcdfb.png">
6. kindly check and confirm, when adding or removing new template, the mappings  should exist.
